payloads/v8/channel.ts

```diff
@@ -930,3 +934,86 @@ export interface APIAllowedMentions {
 	 */
 	replied_user?: boolean;
 }
+
+/**
+ * TODO: Include docs link
+ */
+export interface APIBaseComponent {
+	/**
+	 * The type of the component
+	 */
+	type: ComponentType;
+}
+
+/**
+ * TODO: Include docs link
+ */
+ export const enum ComponentType {
+	/**
+	 * ActionRow component
+	 */
+	ActionRow = 1,
+	/**
+	 * Button component
+	 */
+	Button,
+}
+
+/**
+ * TODO: Include docs link
+ */
+export interface APIActionRowComponent extends APIBaseComponent {
+	/**
+	 * The type of the component
+	 */
+	type: ComponentType.ActionRow;
+	/**
+	 * The compponets in the ActionRow
+	 */
+	components: APIBaseComponent[];
```

Comment:
This should instead be an `APIMessageComponent`, where that is a union of all components (`APIActionRowComponent | APIButtonComponent`)